src/modules/sync-cli.js

```diff
@@ -781,9 +762,84 @@ var self = {
     });
   },
 
-  existingPendingPreHashMap: function(dataset){
-    var pendingPreHashes = dataset.changeHistory || {};
-    return pendingPreHashes;
+  applyPendingChangesToRecords: function(dataset, records){
+    var pendings = dataset.pending;
+    for(var pendingUid in pendings){
```

Comment:
To simplify the logic, I'd remove the for and replace if/else with if.
like here:https://github.com/corinnekrych/fh-ios-sdk/blob/fix-creation-issue.blinking-effect/fh-ios-sdk/Sync/FHSyncDataset.m#L739